packages/devtools/src/encoding.ts

```diff
@@ -32,3 +32,35 @@ export function encodeSignedMap<K extends string>(
 ): Record<K, Array<EncodedDagJWSResult>> {
   return applyMap(data, (records) => records.map(encodeDagJWSResult))
 }
+
+export function decodeEntryCommits(
+  entry: ManagedEntry<EncodedDagJWSResult>
+): ManagedEntry<DagJWSResult> {
+  return { ...entry, commits: entry.commits.map(decodeDagJWSResult) }
+}
+
+export function decodeModel(model: ManagedModel<EncodedDagJWSResult>): ManagedModel<DagJWSResult> {
+  return {
+    schemas: applyMap(model.schemas, (schema) => {
+      return { ...schema, commits: schema.commits.map(decodeDagJWSResult) }
+    }),
+    definitions: applyMap(model.definitions, decodeEntryCommits),
+    tiles: applyMap(model.tiles, decodeEntryCommits),
+  }
```

Comment:
Would it not make more sense to encode the model as follows:
1. put all streams (that are tiles, that's all of them rn) in the *tiles* map
2. make *schemas* and *definition* into an array of strings where we put the streamids

I think this would make the data structure more clear. Anyway just a thought.

Reply:
If we in the future need different commit logs for different networks we could still just maintain one list of definitions and schemas with the suggested model (then having a separate *tiles* map for each network).

Reply:
Yes I was using something like this in the past, but using this structure is more convenient to have type mappings, for example to cast to string as stream IDs here: https://github.com/ceramicstudio/js-idx/blob/master/packages/types/src/index.ts#L26-L36

Reply:
Not sure I follow exactly what that is useful. 

However it seems like now we will have to have a separate *schemas* and *definition* map for each network?

Reply:
We really shouldn't make anything network-specific, it wouldn't be possible anyways.
The point is that datamodels should be portable across networks, so schemas deployed to Clay can be imported, published to a standalone CI node, and then to mainnet for example without having to change anything (as stream and commit IDs are constants).

Reply:
> We really shouldn't make anything network-specific, it wouldn't be possible anyways.

Then we can't allow schemas to be updated at all! I'm pretty sure that supporting differing streams for different networks will be a requirement in the future.

Reply:
I think we had these discussions already in the past when considering the schemas/definitions for IDX and if stream IDs should be network-specific, and every time we opted for making these network-agnostic.
I don't think this should be different for datamodels, especially since it's just a generalization of what we had for IDX. The goal is to make things simpler for developers, if we make this network-specific we're increasing complexity a lot.

Reply:
Yes, the StreamIDs are agnostic, but anchor records are not. Anchor records are also required in order to deal with DIDs that support key rotation. 
My comment is not about increasing complexity, it's about dealing with the complexity that is already present. If we decide not to have an approach for it now that's fine, but we are for sure going to have to deal with it in the future!

Reply:
To me it really seems that anchor commits are a low-level concern that "leak" to the high-level where it shouldn't. If I understand correctly, the problem is really about verifying anchors on different networks?
If so, can't we address the problem at this level? Or have a way to discard invalid anchor commits?

Reply:
> To me it really seems that anchor commits are a low-level concern that "leak" to the high-level where it shouldn't.

It really doesn't leak. What we are doing here is storing the low-level commit log, so it's not really leaking here because we are kind of peering under the hood.

fyi @stbrody this keeps coming up.

Reply:
We could definitely use something else than commits if it's easier, maybe the IPFS archive format Sergey mentioned if that could cover the case?
Fundamentally what we need here is a way to import/export streams like we would in a SQL database or in Mongo with `mongodump` and `mongorestore`. This goes beyond the use case of datamodels I think, for example for people to be able to archive their data and restore it on a personal node for example.

Reply:
This has to do with the security model if the Ceramic stream data structure. Choosing a different data structure is not going to help. Each Ceramic stream will always be confined to specific networks, this will even more be the case once we move to a validator network instead of the anchor service. And in the same way that a contract can have the same address on multiple Eth networks a Ceramic stream can have the same id on different Ceramic networks.

Btw, I'm fine with merging this PR. Just wanted to highlight that thiis is going to be a problem we have to solve at some point in the future.

Reply:
yeah I really can't think of any way we could ever have something like `mongodump`/`mongorestore` for ceramic streams that could be used across ceramic networks while preserving the timestamp history.  You could make one that can import/export the *content* of the stream across networks, but the new stream on the new network would assume that all the commits in the entire stream history happened as of the moment of the import.  This makes sense - you can't import a stream into a new network and expect to import its history into the past.  Probably if we ever had such tool it would compress the stream history and on the new network create only a single commit with the current state, losing the historical versions and timestamps in the process.

Reply:
> Probably if we ever had such tool it would compress the stream history and on the new network create only a single commit with the current state, losing the historical versions and timestamps in the process.

I think this would be problematic as well for DataModels as we use the commit IDs to identity schemas.
Basically what we need is making sure the same IDs can point to the same contents on different networks.

Reply:
@PaulLeCam I'm not sure what you are asking for here. We're saying that on the protocol level it's not possible to maintain the *same* history for streams automatically. You can have a tool on top that maintains a similar commit history, but *commitIds* will differ because we will have different anchor commits for each stream on each network.

Reply:
@oed I wasn't aware Commit IDs are not portable across networks. I think that makes the original IDX design fundamentally flawed, as definitions specify schemas by commit URL.
I don't know what could be the steps forwards here, as the point of datamodels is to be portable across networks.

Reply:
@PaulLeCam Let's discuss this more in depth live. I feel like this thread is not super helpful right now and we could use a more high bandwith discussion. 
I think this PR is fine to merge for now though 👍